third_party/chromium/media/filters/source_buffer_state.cc

```diff
@@ -915,9 +915,43 @@ bool SourceBufferState::OnNewConfigs(
   return success;
 }
 
+#if defined(STARBOARD)
+void SourceBufferState::SetSourceBufferStreamMemoryLimit(size_t limit) {
+  stream_memory_limit_override_ = limit;
+  SetStreamMemoryLimits();
+}
+
+size_t SourceBufferState::GetSourceBufferStreamMemoryLimit() {
+  // a source buffer can be backed my multiple Demuxer streams, although at
+  // YouTube we usually only have a single one. If we've set an override then
+  // all values will be the same, but we shouldn't assume that, so instead we'll
+  // return the largest here if there are multiple.
+  std::vector<size_t> memory_limits;
+  for (const auto& it : audio_streams_) {
+    memory_limits.push_back(it.second->GetStreamMemoryLimit());
+  }
+  for (const auto& it : video_streams_) {
+    memory_limits.push_back(it.second->GetStreamMemoryLimit());
+  }
+
+  if (memory_limits.empty()) return 0;
+
+  return *std::max_element(memory_limits.begin(), memory_limits.end());
+}
+#endif  // defined(STARBOARD)
+
 void SourceBufferState::SetStreamMemoryLimits() {
 #if defined(STARBOARD)
-  // Cobalt doesn't get stream memory limits from the command line.
+  LOG(INFO) << "Custom SourceBuffer size limit="
```

Comment:
nit: can this be on one line?

Reply:
comes to 84 chars if it's on one line - we could trim the log message to fit in but it would become something like "Custom SB memlimit=" to fit. I think a full LOG line is better?

Reply:
Yes, we should keep the LOG as is.  I just realized that my char count plug-in doesn't take space into account, sorry!